pyopencl/array.py

```diff
@@ -2673,6 +2676,17 @@ def if_positive(criterion, then_, else_, out=None, queue=None):
     contains *then_[i]* if *criterion[i]>0*, else *else_[i]*.
     """
 
+    if all(isinstance(k, SCALAR_CLASSES) for k in [criterion,
+                                                   then_,
+                                                   else_]):
```

Comment:
This'll probably be quicker to execute as `isinstance() and isinstance() and isinstance()` (no separate stack frame setup, for instance).

Reply:
9e9571f